src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -195,6 +195,10 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
      *  @param internalListener listener for search request
      */
     protected void checkMaxWorkflows(TimeValue requestTimeOut, Integer maxWorkflow, ActionListener<Boolean> internalListener) {
+        if (!flowFrameworkIndicesHandler.doesIndexExist(CommonValue.GLOBAL_CONTEXT_INDEX)) {
+            internalListener.onResponse(true);
+            return;
```

Comment:
Let's have the rest of the code under `else` rather returning from here. We should persist the listener throughout the method.